jetbrains-ultimate/src/software/aws/toolkits/jetbrains/services/lambda/nodejs/NodeJsLambdaBuilder.kt

```diff
@@ -0,0 +1,51 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.lambda.nodejs
+
+import com.intellij.execution.process.ProcessHandler
+import com.intellij.openapi.application.ReadAction
+import com.intellij.openapi.module.Module
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.roots.ProjectFileIndex
+import com.intellij.openapi.util.io.FileUtil
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.psi.PsiElement
+import software.amazon.awssdk.services.lambda.model.Runtime
+import software.aws.toolkits.jetbrains.services.lambda.BuiltLambda
+import software.aws.toolkits.jetbrains.services.lambda.LambdaBuilder
+import software.aws.toolkits.jetbrains.services.lambda.sam.SamOptions
+import software.aws.toolkits.jetbrains.services.lambda.sam.SamTemplateUtils
+import java.util.concurrent.CompletionStage
+
+// TODO The Lambda Builder logic is the same to Python, we should refactor this a bit.
```

Comment:
What re-factors are necessary? Can we be specific - otherwise it'll be hard for a future person to assess what's outstanding with this TODO.

Reply:
Sure. Basically, we should refactor this a bit to make `getBaseDirectory` method abstract and move the `buildLambda` to super class. I have updated the comment for this.